src/DripsHub.sol

```diff
@@ -471,22 +462,18 @@ abstract contract DripsHub {
     /// @notice Gives funds from the user or their account to the receiver.
     /// The receiver can collect them immediately.
     /// Transfers the funds to be given from the user's wallet to the drips hub contract.
-    /// @param userOrAccount The user or their account
+    /// @param userId The user ID
     /// @param receiver The receiver
     /// @param assetId The used asset ID
     /// @param amt The given amount
     function _give(
-        UserOrAccount memory userOrAccount,
+        uint256 userId,
         address receiver,
         uint256 assetId,
         uint128 amt
-    ) internal {
+    ) internal onlyAccountOwner(userId) {
```

Comment:
I prefer a design, which only uses modifier for `external` and `public` function.
The method is anyway internal, it is easier to understand if all access checks happening and the public function.

Reply:
Me too, but take a look how this function is used, it's only a passthrough implementation of `give` in `ERC20DripsHub` and `EthDripsHub`. If we remove `onlyAccountOwner` here, we'd need to remember to add it in every implementation of `give`, that's poor separation of concerns and a good source of bugs.